cli/app.go

```diff
@@ -226,6 +235,87 @@ var dataTagByFilterFlags = append([]cli.Flag{
 },
 	commonFilterFlags...)
 
+type emptyArgs struct{}
+
+type globalArgs struct {
+	BaseURL string
+	Config  string
+	Debug   bool
+	Quiet   bool
+}
+
+func getValFromContext(name string, ctx *cli.Context) any {
+	// some fuzzy searching is required here, because flags are typically in kebab case, but
+	// params are typically in snake or camel case
+	replacer := strings.NewReplacer("_", "-")
+	dashFormattedName := replacer.Replace(strings.ToLower(name))
+
+	value := ctx.Value(dashFormattedName)
+	if value != nil {
+		return value
+	}
+
+	camelFormattedName := matchAllCap.ReplaceAllString(name, "${1}-${2}")
+	camelFormattedName = strings.ToLower(camelFormattedName)
+
+	return ctx.Value(camelFormattedName)
+}
+
+// TODO(RSDK-9447) - We don't support pointers in this. The problem is that when getting a value
+// from a context for a supported flag, the context will default to populating with the zero value.
+// When getting a value from the context, though, we currently have no way of know if that's going
+// to a concrete value, going to a pointer and should be a nil value, or going to a pointer but should
+// be a pointer to that default value.
```

Comment:
fyi @benjirewis re: our talk about pointers.

Reply:
Sounds good; I'm a little confused about the explanation here but totally fine revisiting this as part of a separate ticket.

Reply:
Lemme know if you wanna chat about it or if any part of this comment is particularly unclear, definitely happy to hop on a call!

Reply:
Thanks! I would say we should meet synchronously if you start doing the work for RSDK-9447.